apps/web/src/hooks/useActiveChainId.ts

```diff
@@ -51,13 +51,14 @@ export const useActiveChainId = () => {
   const chainId = localChainId ?? wagmiChainId ?? (queryChainId >= 0 ? ChainId.BSC : undefined)
 
   const isNotMatched = useDeferredValue(wagmiChainId && localChainId && wagmiChainId !== localChainId)
+  const isWrongNetwork = useMemo(
+    () => Boolean(((wagmiChainId && !isChainSupported(wagmiChainId)) ?? false) || isNotMatched),
+    [wagmiChainId, isNotMatched],
+  )
 
   return {
-    chainId,
-    isWrongNetwork: useMemo(
-      () => Boolean(((wagmiChainId && !isChainSupported(wagmiChainId)) ?? false) || isNotMatched),
-      [wagmiChainId, isNotMatched],
-    ),
+    chainId: isWrongNetwork ? ChainId.BSC : chainId,
```

Comment:
what if it's just not matching? For example the `wagmiChainId` is `eth` while the `localChainId` is `base`

Reply:
btw in what cases this might be an issue?

Reply:
```ts
  const { chainId } = useActiveChainId()
  const provider = usePublicClient({ chainId })
```

when `chainId` is not supported, `usePublicClient` will throw `ChainNotConfiguredError: Chain not configured`

Reply:
> what if it's just not matching? For example the `wagmiChainId` is `eth` while the `localChainId` is `base`

refactor to check if supported only